File: app/src/main/java/com/ryan/common/data/DataSource.java
package com.ryan.common.data;

import com.ryan.androidbaselib.R;
import com.ryan.baselib.util.AppUtils;
import com.ryan.baselib.util.ResourceUtils;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class DataSource {

    public static List<String> getHomeTags() {
        String[] tagArr = ResourceUtils.getStringArray(R.array.array_home_tags);
        return new ArrayList<>(Arrays.asList(tagArr));
    }

    /**
     * UtilsFragment对应的TAG
     * @return list
     */
    public static List<String> getUtilsTags() {
        String[] tagArr = ResourceUtils.getStringArray(R.array.array_utils_tags);
        return new ArrayList<>(Arrays.asList(tagArr));
    }

    /**
     * CustomViewFragment对应的TAG
     * @return list
     */
    public static List<String> getCustomViewTitles() {
        String[] tagArr = ResourceUtils.getStringArray(R.array.array_custom_view_tags);
        return new ArrayList<>(Arrays.asList(tagArr));
    }
}

File: app/src/main/java/com/ryan/common/data/DataSource.kt
package com.ryan.common.data

import com.ryan.androidbaselib.R
import com.ryan.baselib.util.ResourceUtils
import java.util.*
import kotlin.collections.ArrayList

val homeTags: List<String>
    get() {
        val tagArr = ResourceUtils.getStringArray(R.array.array_home_tags)
        return ArrayList(listOf(*tagArr))
    }

/**
 * UtilsFragment对应的TAG
 * @return list
 */
val utilsTags: List<String>
    get() {
        val tagArr = ResourceUtils.getStringArray(R.array.array_utils_tags)
        return ArrayList(listOf(*tagArr))
    }

/**
 * CustomViewFragment对应的TAG
 * @return list
 */
val customViewTitles: List<String>
    get() {
        val tagArr = ResourceUtils.getStringArray(R.array.array_custom_view_tags)
        return ArrayList(listOf(*tagArr))
    }

val toolsTags: List<String>
    get() {
        val tagArr = ResourceUtils.getStringArray(R.array.array_tools_tags)
        return ArrayList(listOf(*tagArr))
    }